### ISSUE TITLE
hands-on evaluation of trusting MFA claims when implementing conditional access authentication strength

### ISSUE THREAD

**Issue by tkol2022** [cisagov/ScubaGear#309]
# 💡 Summary #

We have #80 which evaluates a new Azure AD conditional access strength feature for potential inclusion into the baseline and ScubaGear. We also have #18 which discusses cross-tenant access within the context of device claims

The scope of this issue is to compliment #80, by examining how requiring a specific "authentication strength" for access to the tenant will impact guest users? Will guest users be subject to the same authentication strength requirements (e.g. phishing resistant methods) when a conditional access policy is evaluated for the guest users?

## Implementation notes ##

Setup a guest access scenario:

- Implement a conditional access policy that requires a specific authentication strength
- Create a guest user from another tenant that will be used to test with
- Login as the guest user and access the resource tenant (aka target tenant)

## Document the answer to these questions ##

- [x] Are guest users be subject to the same authentication strength requirements (e.g. phishing resistant methods) when a conditional access policy is evaluated for the guest users?
Answer: Yes, confirmed that a CA policy requiring authentication strength is enforced for guest users.
- [x] Are specific configurations needed to make the integration work between the tenants?
Answer: Although not mandatory, the integration is smoother when the cross-tenant access settings are configured to Trust the MFA from other AAD tenants. This way the guest user performs MFA against their home tenant. If cross-tenant Trust is not configured, the conditional access policy will still enforce authentication strength but the guest user will have to setup an MFA device directly with the resource (target tenant) which may or may not be ideal especially if phishing-resistant MFA is required since that involves physically provisioning FIDO tokens or smartcards for certificate-based authenticaiton.
- [x] Is that configuration documented in the AAD baseline? Is it documented in the right section? If not, create an issue to revise the baseline.
Answer: Currently cross-tenant access settings are mentioned in policy 2.17 which deals with managed devices. I think it might be good to address cross-tenant access with respect to MFA in policy 2.4 since that is centered on MFA configurations. This means we would need to add content to the 2.4 policy.

**Comment by tkol2022** [comment]
## Test Scenario 1 - AAD cross-tenant trust MFA from tenants is Off
This scenario establishes what happens when a guest user arrives at a resource tenant and the AAD > Cross Tenant Access Settings > Trust Settings > Trust multifactor authentication from Azure AD tenants = Disabled.

- I logged into the guest user's home tenant
- Then I tried to access a Teams channel in the resource (target) tenant
- Since the trust MFA is turned off, the system prompted me to register a new MFA device
- This new device, if registered, would be stored in the resource tenant

This test demonstrates that when the Trust setting is disabled, the resource tenant will attempt to have the guest user register an MFA device directly in the resource tenant (instead of the user's home tenant).

**Comment by tkol2022** [comment]
## Test Scenario 2 - AAD trust is On
In this scenario I turned on the Trust multifactor authentication from Azure AD tenants option in the cross-tenant access settings.

- I logged into the guest user's home tenant
- Then I accessed a Teams channel in the resource tenant
- Since the trust MFA is turned On, I was able to access the Teams channel without any further MFA prompts

This scenario demonstrates that since I logged into the home tenant, the MFA claim was re-used when I accessed the resource tenant so there was no need for an additional MFA.

**Comment by tkol2022** [comment]
## Test Scenario 3 - AAD trust is On - Phishing-resistant MFA is required by CA policy
In this scenario the Trust MFA from Azure AD tenants remained on. I also created a new conditional access policy that requires phishing-resistant MFA and scoped it to the test guest user.

- I logged into the guest user's home tenant
- Then I accessed a Teams channel in the resource tenant
- The resource tenant would not let me proceed and I receive a generic "Keep your account secure" error message

This scenario demonstrates that conditional access policies which require a specific MFA authentication strength are enforced on guest users, irrespective of whatever MFA requirements are enforced in the user's home tenant. 
In summary, if we create a new policy to enforce specific types of strong MFA using authentication strength, the policies can be applied to guest users.

![image](https://user-images.githubusercontent.com/107076927/236046306-cbb5267d-42f1-4f64-b37c-9c9fed72928d.png)
